src/Services/MediaService.php

```diff
@@ -68,4 +71,23 @@ public function first(array|int $where = []): ?Model
     {
         return $this->repository->first($where);
     }
+
+    public function createPreview(string $name): Model
+    {
+        $link = Storage::path($name);
+        $name = "preview_{$name}";
+        $dirname = pathinfo($link)['dirname'];
+        $newLink = "{$dirname}/{$name}";
+
+        Image::load($link)
```

Comment:
could we get image content from the argument instead of loading content again?

Reply:
This is not loading the contents of the picture into RAM, but creating the Spatie\Image\Image object. Therefore no.